ci_framework/roles/local_env_vm/tasks/main.yml

```diff
@@ -0,0 +1,137 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Create image directory
+  ansible.builtin.file:
+    path: "{{ cifmw_local_env_vm_basedir }}/images"
+    state: directory
+
+- name: Debug
+  ansible.builtin.debug:
+    msg: "{{ cifmw_local_env_vm_basedir }}/images/{{ cifmw_discovered_image_name }}"
+
+- name: Manage image
+  block:
+    - name: Check image availability
+      register: base_img
+      ansible.builtin.stat:
+        path: "{{ cifmw_local_env_vm_basedir }}/images/{{ cifmw_discovered_image_name }}"
+
+    - name: Fetch image
+      when:
+        - not base_img.stat.exists
+      register: dload_img
+      ansible.builtin.get_url:
+        url: "{{ cifmw_discovered_image_url }}"
+        dest: "{{ cifmw_local_env_vm_basedir }}/images/{{ cifmw_discovered_image_name }}"
+        checksum: "sha256:{{ cifmw_discovered_sha256 }}"
+      until:
+        - dload_img.status_code is defined
+        - dload_img.status_code == 200
+      retries: 2
+      delay: 5
+
+- name: Set some facts for easier usage
+  ansible.builtin.set_fact:
+    cifmw_backing_img: "{{ cifmw_local_env_vm_basedir }}/images/{{ cifmw_discovered_image_name }}"
+    cifmw_work_img: "{{ cifmw_local_env_vm_basedir }}/images/cifmw-vm.qcow2"
+    cifmw_vm_name: "cifmw-vm"
+
+- name: Generate firstboot script as an artifact
+  ansible.builtin.copy:
+    dest: "{{ cifmw_local_env_vm_basedir }}/artifacts/firstboot.sh"
+    src: "firstboot.sh"
+    mode: 0755
+
+- name: Create image layer
+  register: create_layer
+  ansible.builtin.command:
+    creates: "{{ cifmw_work_img }}"
+    cmd: >-
+      qemu-img create -o backing_file={{ cifmw_backing_img }},backing_fmt=qcow2
+      -f qcow2 {{ cifmw_work_img }} {{ cifmw_local_env_vm_space }}
+
+- name: Prepare the workload image
+  when:
+    - create_layer is changed
+  ansible.builtin.shell:
+    executable: /bin/bash
+    cmd: >-
+      virt-sysprep -a {{ cifmw_work_img }} --network --update --selinux-relabel
+      --ssh-inject "root:file:{{ ansible_user_dir }}/.ssh/authorized_keys"
```

Comment:
The user may not have an  authorized_keys file. If we assume that we need to document it, and ensure that the quick start guide covers it.

Reply:
I think another solution would be to add a pre-run task that checks if the file is present, if not, throw an error prompting the user to create it.

Reply:
hmm yeah. that would allow to support any ssh format - for instance, I don't have RSA anymore ;). Lemme amend in order to add some pre in the playbook.